appium/webdriver/webdriver.py

```diff
@@ -497,9 +500,33 @@ def orientation(self, value: str) -> None:
         if value.upper() in allowed_values:
             self.execute(Command.SET_SCREEN_ORIENTATION, {'orientation': value})
         else:
-
             raise WebDriverException("You can only set the orientation to 'LANDSCAPE' and 'PORTRAIT'")
 
+    def assert_extension_exists(self, ext_name: str) -> 'WebDriver':
+        """
+        Verifies if the given extension is not present in the list of absent extensions
+        for the given driver instance.
+        This API is designed for private usage.
+
+        :param ext_name: extension name
+        :return: self instance for chaining
+        :raise UnknownMethodException: If the extension has been marked as absent once
+        """
+        if ext_name in self._absent_extensions:
+            raise UnknownMethodException()
+        return self
+
+    def mark_extension_absence(self, ext_name: str) -> 'WebDriver':
+        """
+        Marks the given extension as absent for the given driver instance.
+        This API is designed for private usage.
+
+        :param ext_name: extension name
+        :return: self instance for chaining
+        """
+        self._absent_extensions.add(ext_name)
```

Comment:
Can we leave a log here? Perhaps... debug or info?

The main usage is for us. When we get reports by users even if they use the latest driver versions, we maybe should add new extension commands for drivers. Such logs will help to find them.

Reply:
added a log